sonar-ws/src/main/java/org/sonarqube/ws/client/HttpConnector.java

```diff
@@ -192,16 +192,20 @@ private WsResponse post(PostRequest postRequest) {
     return okHttpRequestBuilder;
   }
 
-  private HttpResponse doCall(Request okRequest) {
+  private OkHttpResponse doCall(Request okRequest) {
     Call call = okHttpClient.newCall(okRequest);
     try {
       Response okResponse = call.execute();
-      return new HttpResponse(okResponse);
+      return new OkHttpResponse(okResponse);
     } catch (IOException e) {
       throw new IllegalStateException("Fail to request " + okRequest.urlString(), e);
     }
   }
 
+  public static Builder builder() {
```

Comment:
if you provide a static factory method to create a builder:

- it would remove an ambiguity to call it `newBuilder` (returns a new instance for each call)
- constructor of Builder class should be private to enforce use of the static factory method